x/distribution/keeper/delegation.go

```diff
@@ -64,6 +64,13 @@ func (k Keeper) calculateDelegationRewards(ctx sdk.Context, val sdk.Validator, d
 	startingPeriod := startingInfo.PreviousPeriod
 	stake := startingInfo.Stake
 
+	// due to rounding differences due to order of operations, stake calculated
```

Comment:
I don't quite understand this. We already set initial stake with the truncated delegator share exchange rate - precisely which order of operations difference is causing a rounding error? An example would be :100:

Reply:
k I can expand the comment - the truncation does not sufficiently account for the difference aka 
aka `1000 * (999 * 0.001) != (1000 * 999) * 0.001` depending on how many decimal places we use in decimal 

basically currentStake and currentStakeF1 is the same calculation but with rearranged variables. In F1 we round in one way due to the event fraction - in staking we round in another way using `GetDelegatorShareExRate`

Reply:
Shouldn't there be a way to truncate so that `1000 * (999 * 0.001) <= (1000 * 999) * 0.001` (or vice versa depending on what we want)?

Reply:
I think `1000 * (999 * 0.001) <= (1000 * 999) * 0.001` is actually already true the problem is is that the RS of this equation is distribution `stake` calculation hence it can be larger than what is presented by the staking calculation.. Cleanest solution IMO which would remove the use of the correction factor is to change the order of operations in staking to correspond to what's happening in distribution here

Reply:
OK, that does sound cleaner, let's try that.

Reply:
implemented - it works